extra_streamlit_components/CookieManager/__init__.py

```diff
@@ -63,17 +68,21 @@ def set(
         }
         # Remove None's
         options = {k: v for k, v in options.items() if v is not None}
-        did_add = self.cookie_manager(method="set", cookie=cookie, value=val, options=options, key=key, default=False)
-        if did_add:
-            self.cookies[cookie] = val
+        try:
+            did_add = self.cookie_manager(method="set", cookie=cookie, val=val, options=options, key=key, default=False) 
+            return did_add
+        except:
+            return False
 
     def delete(self, cookie, key="delete"):
         if cookie is None or cookie == "":
             return
-        did_add = self.cookie_manager(method="delete", cookie=cookie, key=key, default=False)
-        if did_add:
-            del self.cookies[cookie]
+        try:
+            did_add = self.cookie_manager(method="delete", cookie=cookie, key=key, default=False)
+            return did_add
```

Comment:
maybe rename to did_del.
```suggestion
            did_del = self.cookie_manager(method="delete", cookie=cookie, key=key, default=False)
            return did_del
```

Reply:
Will make this change too.